app/Http/Controllers/ViewAssetsController.php

```diff
@@ -80,190 +82,116 @@ public function getRequestItem($itemType, $itemId = null)
     {
         $item = null;
         $fullItemType = 'App\\Models\\' . studly_case($itemType);
+
         if ($itemType == "asset_model") {
             $itemType = "model";
         }
         $item = call_user_func(array($fullItemType, 'find'), $itemId);
+
         $user = Auth::user();
-        $quantity = $data['item_quantity'] = Input::has('request-quantity') ? e(Input::get('request-quantity')) : 1;
+
 
         $logaction = new Actionlog();
         $logaction->item_id = $data['asset_id'] = $item->id;
         $logaction->item_type = $fullItemType;
         $logaction->created_at = $data['requested_date'] = date("Y-m-d H:i:s");
+
         if ($user->location_id) {
             $logaction->location_id = $user->location_id;
         }
         $logaction->target_id = $data['user_id'] = Auth::user()->id;
         $logaction->target_type = User::class;
 
+        $data['item_quantity'] = Input::has('request-quantity') ? e(Input::get('request-quantity')) : 1;
         $data['requested_by'] = $user->present()->fullName();
-        $data['item_name'] = $item->name;
+        $data['item'] = $item;
         $data['item_type'] = $itemType;
+        $data['target'] = Auth::user();
+
 
         if ($fullItemType == Asset::class) {
             $data['item_url'] = route('hardware.show', $item->id);
-            $slackMessage = ' Asset <'.url('/').'/hardware/'.$item->id.'/view'.'|'.$item->present()->name().'> requested by <'.url('/').'/users/'.$item->user_id.'/view'.'|'.$user->present()->fullName().'>.';
         } else {
             $data['item_url'] = route("view/${itemType}", $item->id);
-            $slackMessage = $quantity. ' ' . class_basename(strtoupper($logaction->item_type)).' <'.$data['item_url'].'|'.$item->name.'> requested by <'.url('/').'/user/'.$item->id.'/view'.'|'.$user->present()->fullName().'>.';
-        }
-
-        $settings = Setting::getSettings();
 
-        if ($settings->slack_endpoint) {
```

Comment:
The cleanup happening here is just gorgeous. All this complexity melting away into some very simple string interpolations. *BEAUTIFUL*!